app/controllers/proxy_controller.rb

```diff
@@ -31,4 +29,18 @@ def index
 
     render :json => response.body, :status => response.code
   end
+
+  private
+
+  def slice_eholdings_namespace
```

Comment:
How about just

```
def rmapi_path
  request.fullpath.slice('/eholdings')
end
```

And then just use that in the template string directly.

Reply:
this is what i thought you meant in your above comment.  it breaks us free of the unnecessary `before_action`.  it needs a slight tweak from how you've written it (`.slice` returns the sliced segment not the remaining segment) but this is the right idea for sure.